src/plugins/stake-plugins/thorchainSavers/tcSaversPlugin.ts

```diff
@@ -317,90 +306,92 @@ const getStakePosition = async (opts: EdgeGuiPluginOptions, request: StakePositi
-  }
-  const poolsJson = await poolsResponse.json()
-  pools = asPools(poolsJson)
+async function fetchSavers(opts: EdgeGuiPluginOptions, asset: string): Promise<Saver[]> {
+  const { ninerealmsClientId } = asInitOptions(opts.initOptions)
+  const response = await fetchWaterfall(thornodeServers, `thorchain/pool/${asset}/savers`, {
+    headers: { 'x-client-id': ninerealmsClientId }
+  })
 
-  return {
-    asset,
-    currencyCode,
-    primaryAddress,
-    pluginId,
-    pools,
-    savers,
-    wallet
+  if (!response.ok) {
+    const responseText = await response.text()
+    throw new Error(`Thorchain could not fetch /pool/savers: ${responseText}`)
   }
+  const saversJson = await response.json()
+  return asSavers(saversJson)
 }
 
-const getStakePositionWithPools = async (params: StakePositionWithPoolsParams): Promise<StakePosition> => {
-  const { asset, currencyCode, primaryAddress, pluginId, pools, savers, wallet } = params
-  const saver = params.saver ?? savers.find(s => s.asset_address.toLowerCase() === primaryAddress.toLowerCase())
-  const pool = pools.find(p => p.asset === asset)
-  let stakedAmount = '0'
-  let earnedAmount = '0'
-  if (saver != null && pool != null) {
-    const { units, asset_deposit_value: assetDepositValue } = saver
-    const { saversDepth, saversUnits } = pool
-    stakedAmount = assetDepositValue
-    const redeemableValue = div(mul(units, saversDepth), saversUnits, DIVIDE_PRECISION)
-    earnedAmount = sub(redeemableValue, stakedAmount)
-
-    // Convert from Thor units to exchangeAmount
-    stakedAmount = div(stakedAmount, THOR_LIMIT_UNITS, DIVIDE_PRECISION)
-    earnedAmount = div(earnedAmount, THOR_LIMIT_UNITS, DIVIDE_PRECISION)
-
-    // Convert from exchangeAmount to nativeAmount
-    const multiplier = getCurrencyCodeMultiplier(wallet.currencyConfig, currencyCode)
-    stakedAmount = mul(stakedAmount, multiplier)
-    earnedAmount = mul(earnedAmount, multiplier)
-
-    // Truncate decimals
-    stakedAmount = toFixed(stakedAmount, 0, 0)
-    earnedAmount = toFixed(earnedAmount, 0, 0)
-
-    // Cap negative value to 0
-    earnedAmount = max(earnedAmount, '0')
+function saverToPosition(currencyConfig: EdgeCurrencyConfig, currencyCode: string, saver: Saver, pool: Pool): StakePosition {
+  const pluginId = currencyConfig.currencyInfo.pluginId
+
+  const multiplier = getCurrencyCodeMultiplier(currencyConfig, currencyCode)
+  function thorToNative(amount: string): string {
+    return toFixed(mul(div(amount, THOR_LIMIT_UNITS, DIVIDE_PRECISION), multiplier), 0, 0)
   }
 
+  const { units, asset_deposit_value: assetDepositValue } = saver
+  const stakedAmount = thorToNative(assetDepositValue)
   const canUnstake = gt(stakedAmount, '0')
 
+  const { saversDepth, saversUnits } = pool
+  const redeemableValue = div(mul(units, saversDepth), saversUnits, DIVIDE_PRECISION)
+  const earnedThorAmount = max(sub(redeemableValue, assetDepositValue), '0')
```

Comment:
Now that the other consts use this nice `thorToNative` to clearly describe the semantic behind the const value, I kinda want to know what's going on with these two. Maybe some comments would be nice to explain all the operations.

Reply:
I have no idea. These seem to be in "Thor units", but I don't understand the redeemable value calculation at all. The earned value seems to be the difference between what we deposited and what we can redeem, which makes sense. The `max` ensures our earned amount can't be negative.

My only goal here was to preserve the logic as it was, without any changes.

Reply:
No change needed then. If you want, you can add a WTF comment like "// TODO: Describe what this math is mathing"